arangod/Aql/OptimizerRules.cpp

```diff
@@ -7471,6 +7470,29 @@ void arangodb::aql::geoIndexRule(Optimizer* opt,
     LimitNode* limit = nullptr;
     bool canUseSortLimit = true;
 
+    bool mustRespectIdxHint = false;
+    if (plan->hasForcedIndexHints()) {
```

Comment:
This is not correct. We should not care here if the **plan** has any forced index hints, but if the **FOR** loop we are looking at has forced index hints.